src/Service/ProgrammesAggregationService.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Service;
+
+use BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository\ClipRepository;
+use BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository\CoreEntityRepository;
+use BBC\ProgrammesPagesService\Domain\Entity\CoreEntity;
+use BBC\ProgrammesPagesService\Domain\Entity\Programme;
+use BBC\ProgrammesPagesService\Mapper\ProgrammesDbToDomain\CoreEntityMapper;
+use BBC\ProgrammesPagesService\Cache\CacheInterface;
+use InvalidArgumentException;
+
+class ProgrammesAggregationService extends AbstractService
+{
+    const AGGREGATION_VALID_TYPES = [
+        'Series',
+        'Episode',
+        'Clip',
+        'Franchise',
+        'Gallery',
+        'Collection',
+        'Season',
+    ];
+
+    public function __construct(
+        CoreEntityRepository $repository,
+        CoreEntityMapper $mapper,
+        CacheInterface $cache
+    ) {
+        parent::__construct($repository, $mapper, $cache);
+    }
+
+    /**
+     * @return Series[]
+     */
+    public function findDescendantSeries(
+        Programme $programme,
+        ?int $limit = self::DEFAULT_LIMIT,
+        int $page = self::DEFAULT_PAGE
+    ) : array {
+        return $this->findDescendantsByType($programme, 'Series', $limit, $page);
+    }
+
+    /**
+     * @return Episode[]
+     */
+    public function findDescendantEpisodes(
+        Programme $programme,
+        ?int $limit = self::DEFAULT_LIMIT,
+        int $page = self::DEFAULT_PAGE
+    ) : array {
+        return $this->findDescendantsByType($programme, 'Episode', $limit, $page);
+    }
+
+    /**
+     * @return Clip[]
+     */
+    public function findDescendantClips(
+        Programme $programme,
+        ?int $limit = self::DEFAULT_LIMIT,
+        int $page = self::DEFAULT_PAGE
+    ) : array {
+        return $this->findDescendantsByType($programme, 'Clip', $limit, $page);
+    }
+
+    /**
+     * @return Gallery[]
+     */
+    public function findDescendantGalleries(
+        Programme $programme,
+        ?int $limit = self::DEFAULT_LIMIT,
+        int $page = self::DEFAULT_PAGE
+    ) : array {
+        return $this->findDescendantsByType($programme, 'Gallery', $limit, $page);
+    }
+
+    /**
+     * @return CoreEntity[]
+     */
+    private function findDescendantsByType(
+        Programme $programme,
+        string $type,
+        ?int $limit,
+        int $page,
+        $ttl = CacheInterface::NORMAL
+    ) : array {
+        $this->assertEntityType($type, self::AGGREGATION_VALID_TYPES);
```

Comment:
Do you need to pass `self::AGGREGATION_VALID_TYPES` in? Can't you just reference it in `assertEntityType()`?